src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -253,6 +265,12 @@ public Map<Keyspace, Integer> recoverFiles(ColumnFamilyStore.FlushReason flushRe
     {
         CommitLogReplayer replayer = CommitLogReplayer.construct(this, getLocalHostId());
         replayer.replayFiles(clogs);
+
+        /*we need to reload segments with invalid mutations as it may have been updated by the commit log replayer
```

Comment:
I don't quite get this comment: AFAIU, the following line is required simply in order to retrieve the invalid mutations and check them above, what am I missing?

Reply:
I wrote it in this way because initially I was only incrementing this list. But then I figured that contents can also be removed, so we need a full reload. But yeah you're right - I'll simplify this :)